MyLibrary/Tests/MyLibraryTests/MyLibraryTests.swift

```diff
@@ -68,5 +69,38 @@ final class MyLibraryTests: XCTestCase {
         // Then
         XCTAssertNil(isLuckyNumber)
     }
+    
+    func testWeatherModelForValidTemperature() {
+        let decoder = JSONDecoder()
+        let data = Data("{\"main\": {\"temp\":58}}".utf8)
+        do {
+            let container = try decoder.decode(Weather.self, from: data)
+            XCTAssertNotNil(container)
+            XCTAssert(container.main.temp == 58.0)
+        } catch {
+            XCTFail("Failed to decode temp from json")
+        }
+    }
+    func testWeatherModelForInvalidTemperature() {
+        let decoder = JSONDecoder()
+        let data = Data("{\"main\": {\"temp\":\"invalid temp\"}}".utf8)
+        do {
+            let _ = try decoder.decode(Weather.self, from: data)
+            XCTFail("Should not have been possible to decode temp for invalid input")
+        } catch {
+            XCTAssert(true == true)
+        }
+    }
+    
+    // Integration Tests
+    func testWeatherServiceReturnsCorrectTemperature() async {
```

Comment:
If you put the integration tests in the same test suite (and here they are part of the same test file to boot), then you won't be able to launch integration tests up on the CI without explicitly disabling this test in the scheme.

So, instead, I would recommend creating a separate test suite.  You can create a second unit-test suite and rename it to "Integration Tests".  I would then see how to trigger this or that test suite from the command line.